### PR TITLE
Sync vehicles destruction

### PR DESCRIPTION
**Pull request by tornac1234**
- [x] Sync vehicle destruction
- [x] Fix cyclops destruction not being synced
- [x] Sync cyclops destroy(/restore) command(s)
   - [x] Sync destroy command
   - ~~[ ] Fix restore command~~ (As discussed, it's too much work for something not worth it)
   - [x] Make the commands only apply to the cyclops we're inside (instead it applies to all cyclops)

### CODE REVIEW

**Comment by sarayuuki** [comment]
Is there anyway to merge this pull request locally? I would like to test out the changes.

**Comment by RivalsRebels** [comment]
> Is there anyway to merge this pull request locally? I would like to test out the changes.

You can go on Tornac's repo, and bulid Nitrox from there

**Comment by dartasen** [comment]
> Is there anyway to merge this pull request locally? I would like to test out the changes.

You can if you're familiar with git. But I recommand to wait until we've and merged the PR.

**Comment by sarayuuki** [comment]
Ok thanks for the answer

**Comment by tornac1234** [comment]
I forgot to send the latest commit ☠

File: NitroxPatcher/Patches/Dynamic/CyclopsDestructionEvent_OnConsoleCommand_Patch.cs
Comment (dartasen) [inline NitroxPatcher/Patches/Dynamic/CyclopsDestructionEvent_OnConsoleCommand_Patch.cs:21]: There is already a translated error message for those cases, you can have a look at speed, sub console commands patches

File: NitroxServer/Communication/Packets/Processors/VehicleDestroyedPacketProcessor.cs
Comment (Measurity) [inline NitroxServer/Communication/Packets/Processors/VehicleDestroyedPacketProcessor.cs:25]: Is it intended to not remove ownership after destroy in `CyclopsDestroyedProcessor.Process`?

**Review by dartasen — APPROVED** [review]

**Pull request merged**
